diff --git a/build.sh b/build.sh
index fb74328..de8251a 100755
--- a/build.sh
+++ b/build.sh
@@ -138,6 +138,10 @@ function build_protobuf() {
 }
 
 function build_nccl() {
+    local nccl_debug=0
+    if [ "$FLUX_DEBUG" == "ON" ]; then
+        nccl_debug=1
+    fi
     pushd $NCCL_ROOT
     export BUILDDIR=${NCCL_ROOT}/build
     export PREFIX=${BUILDDIR}/local
@@ -149,9 +153,9 @@ function build_nccl() {
         for arch in "${arch_list[@]}"; do
             NCCL_COMPILE_OPTIONS_ARCH="-gencode=arch=compute_${arch},code=sm_${arch} ${NCCL_COMPILE_OPTIONS_ARCH}"
         done
-        make -j${nproc} src.staticlib NVCC_GENCODE="${NCCL_COMPILE_OPTIONS_ARCH}" VERBOSE=1
+        make -j${JOBS} src.staticlib DEBUG=${nccl_debug} NVCC_GENCODE="${NCCL_COMPILE_OPTIONS_ARCH}" VERBOSE=1
     else
-        make -j${nproc} src.staticlib VERBOSE=1
+        make -j${JOBS} src.staticlib DEBUG=${nccl_debug} VERBOSE=1
     fi
     # only install static lib
     mkdir -p ${PREFIX}/lib
diff --git a/src/coll/CMakeLists.txt b/src/coll/CMakeLists.txt
index 91424dc..a59292c 100644
--- a/src/coll/CMakeLists.txt
+++ b/src/coll/CMakeLists.txt
@@ -21,7 +21,7 @@ list(APPEND FLUX_CUDA_OP_TARGETS ${LIB_NAME})
 set(FLUX_CUDA_OP_TARGETS ${FLUX_CUDA_OP_TARGETS} PARENT_SCOPE)
 if (BUILD_THS)
   file(GLOB THS_FILES ths_op/*.cc)
-  file(GLOB ALL2ALL_THS_FILES ths_op/dis_scatter*.cc ths_op/all2all_op.cc ths_op/all2all_single_2d.cc)
+  file(GLOB ALL2ALL_THS_FILES ths_op/dis_scatter*.cc ths_op/all2all_op.cc ths_op/all2all_single_2d.cc ths_op/isendrecv.cc)
   list(REMOVE_ITEM THS_FILES ${ALL2ALL_THS_FILES})
   if(ENABLE_NVSHMEM)
     list(APPEND THS_FILES ${ALL2ALL_THS_FILES})
